[BUGFIX release] Remove requirement for jquery in Octane

pzuraq opened the pull request:
Based on community feedback and early testing, the core team
has decided that disabling jquery-integration does not need to be a
requirement for enabling Octane mode, so we can remove it from
the error messages.

rwjblue commented on lib/index.js:
```diff
@@ -79,12 +79,6 @@ module.exports = {
         );
       }
 
-      if (optionalFeaturesMissing || optionalFeatures.isFeatureEnabled('jquery-integration')) {
```
Instead of removing this, we should ensure that the flag is set (not `undefined`).

Commit:
[BUGFIX] Removes requirement for disabling Jquery in Octane

Removes the requirement for disabling Jquery, instead only requiring
that users explicitly enable/disable it.
